lib/process-data-availability.js

```diff
@@ -454,82 +521,145 @@ const processDataAvailability = async (availabilityReport) => {
   transformed.lookups = lookups;
   transformed.lookupValues = processedLookupValues;
   transformed.availability.resourcesBinary = resourcesBinary;
-  transformed.availability.resources = computeResourceAvailability(
-    availabilityAggregatesByResource
-  );
+  transformed.availability.resources = computeResourceAvailability(availabilityAggregatesByResource);
 
   return { ...rest, ...transformed };
 };
 
+const updateResourceBinaryAvailability = ({ resourceName, type, statType, availability, resourcesBinary, parentResourceName } = {}) => {
+  if (parentResourceName) {
+    resourcesBinary[parentResourceName].expansions[resourceName][type][statType] = computeBins(
+      availability,
+      resourcesBinary[parentResourceName].expansions[resourceName][type][statType]
+    );
+  } else {
+    resourcesBinary[resourceName][type][statType] = computeBins(availability, resourcesBinary[resourceName][type][statType]);
+  }
+};
+
+/**
+ *
+ * @param {Object} obj
+ * @param {String} obj.resourceName
```

Comment:
For documentation, we only need to break out the properties if they're not obvious (from the name). Otherwise we can see what they are from the destructuring. Can just leave as `@param {Object} params description of params` etc.

Reply:
One reason for using these JSDoc types was that we eventually wanted our libraries to be in Typescript. Using JSDoc types it would be much easier to migrate over when we do.